# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*



Lane: Refresh / Content Opportunity Scoring

Why:
I selected the Refresh / Content Opportunity Scoring lane because content teams often have limited time and cannot manually review every page. This project aims to build a ranking system that helps prioritize which pages should be reviewed first for refresh, expansion, protection, pruning, or monitoring using observable search and engagement signals. The goal is to support better decision-making rather than simply train a machine learning model. This lane also aligns with the internship objective of producing evidence-based ranked recommendations.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

Research Question

Which content pages should be reviewed first for refresh, expansion, protection, pruning, or monitoring based on historical search and engagement signals?

Decision

Which pages should the SEO or content team prioritize for review?

Who acts?

SEO managers or content editors.

Action

Review the highest-ranked pages and decide whether to refresh, expand, protect, prune, or monitor them.

Cost of a Wrong Recommendation
False Positive: A healthy page is recommended for review, wasting editor time and review capacity.
False Negative: A page that truly needs attention is missed, allowing traffic or engagement to continue declining.

Why Machine Learning?

Page performance depends on many interacting signals such as impressions, clicks, CTR, average position, engagement rate, freshness, and content characteristics. These relationships are difficult to capture using a single rule, so machine learning can help prioritize pages for human review.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [5]:
import pandas as pd

# Load the starter dataset
df = pd.read_csv("../../data/raw/content_refresh_anonymized.csv")

# -------------------------------
# Basic Information
# -------------------------------
print("=" * 60)
print("DATASET OVERVIEW")
print("=" * 60)
print(f"Rows: {len(df):,}")
print(f"Columns: {len(df.columns)}")
print(f"Unique clients: {df['client_id'].nunique()}")

# -------------------------------
# Content Type Distribution
# -------------------------------
print("\n" + "=" * 60)
print("CONTENT TYPES")
print("=" * 60)
print(df["content_type"].value_counts())

# -------------------------------
# Trend Direction Distribution
# -------------------------------
print("\n" + "=" * 60)
print("TREND DIRECTION")
print("=" * 60)
print(df["trend_direction"].value_counts())

print("\nTrend Direction (%)")
print((df["trend_direction"].value_counts(normalize=True) * 100).round(2))

# -------------------------------
# Performance Statistics
# -------------------------------
print("\n" + "=" * 60)
print("PERFORMANCE STATISTICS")
print("=" * 60)

print(f"Average 90-day Impressions : {df['impressions_90d'].mean():.2f}")
print(f"Average 90-day Clicks      : {df['clicks_90d'].mean():.2f}")
print(f"Average CTR (%)            : {df['ctr'].mean():.2f}")

# Average Position (0 = no data)
avg_position = df["avg_position"].replace(0, pd.NA).mean()
print(f"Average Search Position    : {avg_position:.2f}")

print(f"Average Engagement Rate (%) : {df['engagement_rate'].mean():.2f}")
print(f"Average AI Traffic (%)      : {df['ai_traffic_pct'].mean():.2f}")

# -------------------------------
# Missing Values
# -------------------------------
print("\n" + "=" * 60)
print("MISSING VALUES")
print("=" * 60)
print(df.isnull().sum().sort_values(ascending=False).head(10))

DATASET OVERVIEW
Rows: 30,000
Columns: 44
Unique clients: 32

CONTENT TYPES
content_type
keyword article       27207
feedly article         2096
comparison article      697
Name: count, dtype: int64

TREND DIRECTION
trend_direction
down      16262
stable     5962
up         4388
new        2236
flat       1152
Name: count, dtype: int64

Trend Direction (%)
trend_direction
down      54.21
stable    19.87
up        14.63
new        7.45
flat       3.84
Name: proportion, dtype: float64

PERFORMANCE STATISTICS
Average 90-day Impressions : 5200.37
Average 90-day Clicks      : 16.10
Average CTR (%)            : 0.51
Average Search Position    : 17.03
Average Engagement Rate (%) : 2.53
Average AI Traffic (%)      : 0.77

MISSING VALUES
provider_used        21438
word_count            7699
char_count            7699
word_count_tier       7699
char_count_tier       7699
model_used            5733
trend_pct             3388
competition_level     2610
search_volume         2468
cpc               

The starter dataset contains 30,000 content pages described by 44 features across 32 pseudonymized clients. Most pages (27,207, about 90.7%) are keyword articles, while 2,096 are Feedly articles and 697 are comparison articles. The dataset also shows substantial variation in page performance: 16,262 pages are currently marked as "down" in trend_direction, indicating many pages may benefit from review. These observations suggest that ranking pages by their likelihood of needing attention is a meaningful problem for content teams, making the Refresh / Content Opportunity Scoring lane worth investigating over the next seven weeks.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

This project aims to provide decision support for SEO managers and content editors by ranking pages that appear to deserve review based on historical search and engagement signals. The model will identify pages that may be good candidates for refresh, expansion, protection, pruning, or monitoring. However, it will not prove that refreshing a page will improve its performance or establish a causal relationship between any feature and future traffic. The recommendations are observational and directional, and final decisions should remain with human experts.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.